src/java.net.http/share/classes/jdk/internal/net/http/websocket/Frame.java

```diff
@@ -276,7 +288,10 @@ HeaderWriter mask(int value) {
         }
 
         HeaderWriter noMask() {
-            firstChar &= ~0b00000000_10000000;
+            // The negation "~" sets the high order bits
```

Comment:
Stamping this in front of every of the four closely sitting casts seems excessive:

                // The negation "~" sets the high order bits
                // so the value is more than 16 bits and the
                // compiler will emit a warning if not cast

Reply:
I don't disagree. I had the same feeling. But I don't necessarily read a file from top to bottom - and someone just glancing at one of these methods might wonder why one line has the cast and the other hasn't.

Reply:
Readers should always consider as much of the context as practical; but I like your solution.